src/main/java/com/aws/greengrass/mqtt/bridge/clients/LocalMqtt5Client.java

```diff
@@ -417,9 +439,7 @@ void subscribe(String topic) throws RetryableMqttOperationException {
                         topic,
                         QOS.AT_LEAST_ONCE,
                         isNoLocal(topic),
-                        // always set retainAsPublished so we have the retain flag available,
-                        // when we bridge messages, we'll set retain flag based on user route configuration.
-                        true,
+                        retainAsPublished(topic),
```

Comment:
Do not do this. Read the comment.

Reply:
In that case do we need to read the retainAsPublished flag from the config in the local client at all? Might have misunderstood this description [https://sim.amazon.com/issues/GG-42541](https://sim.amazon.com/issues/GG-42541)

Reply:
No you do not need this flag to be read in the client.

Reply:
It needs to be used only when _routing_ the message to the destination (iot core)